python/packages/zarathustra/skills/subgraph_query_abci/behaviours.py

```diff
@@ -175,21 +216,28 @@ def async_act(self) -> Generator:
 
         with self.context.benchmark_tool.measure(self.behaviour_id).local():
             sender = self.context.agent_address
-            config = self._check_config(self.params.config)
-            subgraph_url = self.params.config["subgraph_url"]
-            subgraph_query = self.params.config["subgraph_query"]
-            content = json.dumps({"subgraph_url": subgraph_url, "subgraph_query": subgraph_query})
-            self.context.logger.info(f"subgraph config: {content}")
+            subgraph_queries = self._verify_subgraph_config(self.params.config)
+            self.context.logger.info(f"subgraph queries: {subgraph_queries}")
+            content = serialize({"subgraph_queries": subgraph_queries})
             payload = LoadSubgraphComponentsPayload(sender=sender, content=content)
 
         with self.context.benchmark_tool.measure(self.behaviour_id).consensus():
             yield from self.send_a2a_transaction(payload)
             yield from self.wait_until_round_end()
 
-    def _check_config(self, config):
-        required = ("subgraph_api_key", "subgraph_url", "subgraph_query")
-        if not all(map(config.get, required)):
-            raise ValueError(f"Ensure all required parameters are provided: {required}")
+    def _verify_subgraph_config(self, config: dict) -> dict[str, SubgraphQueryConfig]:
+        if not config.get("subgraph_api_key"):
+            self.context.logger.warning("No subgraph_api_key provided!")
+
+        if not (subgraph_queries := config.get("subgraph_queries")):
+            raise ValueError("No subgraph queries provided in aea-config.yaml")
+
+        required = ("url", "chain", "queries")
+        for subgraph_name, config in subgraph_queries.items():
+            if not all(map(config.get, required)):
+                raise ValueError(f"Ensure all required fields are provided for {subgraph_name}: {required}")
+
+        return {name: SubgraphQueryConfig(**data) for name, data in subgraph_queries.items()}
```

Comment:
these are effectively 3 checks. 
1. First is that subgraph_queries must exist (else this AbciApp doesn't make any sense, and neither can it operate without)
2. Second, the subgraph_queries is a mapping, whose values should consist of an `url`, `chain` and a list of `queries`
3. The `SubgraphQueryConfig` is a pydantic model that will throw if the datatype doesn't match the specification.